controllers/dpa_controller.go

```diff
@@ -136,14 +140,69 @@ func (r *DPAReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Owns(&velerov1.VolumeSnapshotLocation{}).
 		Owns(&appsv1.DaemonSet{}).
 		Owns(&security.SecurityContextConstraints{}).
-		Owns(&corev1.Secret{}).
 		Owns(&corev1.Service{}).
 		Owns(&routev1.Route{}).
 		Owns(&corev1.ConfigMap{}).
+		Watches(&source.Kind{Type: &corev1.Secret{}}, &labelHandler{}).
 		WithEventFilter(veleroPredicate(r.Scheme)).
 		Complete(r)
 }
 
+type labelHandler struct {
+}
+
+func (l *labelHandler) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+	// check for the label & add it to the queue
+	namespace := evt.Object.GetNamespace()
+	dpaname := evt.Object.GetLabels()[namespace+".dataprotectionapplication"]
+	if evt.Object.GetLabels()[oadpv1alpha1.OadpOperatorLabel] == "" || dpaname == "" {
+		return
+	}
+
+	q.Add(reconcile.Request{NamespacedName: types.NamespacedName{
+		Name:      dpaname,
+		Namespace: namespace,
+	}})
+
+}
+func (l *labelHandler) Delete(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
+
+	namespace := evt.Object.GetNamespace()
+	dpaname := evt.Object.GetLabels()[namespace+".dataprotectionapplication"]
+	if evt.Object.GetLabels()[oadpv1alpha1.OadpOperatorLabel] == "" || dpaname == "" {
+		return
+	}
+	q.Add(reconcile.Request{NamespacedName: types.NamespacedName{
+		Name:      dpaname,
+		Namespace: namespace,
+	}})
+
+}
+func (l *labelHandler) Update(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+
+	if evt.ObjectNew.GetLabels()[oadpv1alpha1.OadpOperatorLabel] != "" {
```

Comment:
add ```if evt.Object.GetLabels()[oadpv1alpha1.OadpOperatorLabel] == "" || dpaname == "" {
 		return
 	}
 	```

Reply:
Updated it. I had to use `ObjectNew` instead of `Object`. Update function supports only `ObjectNew` & `ObjectOld`. Let me know if I have to add a case for `ObjectOld` as well :)

Reply:
Nope I think ObjectNew is the only thing we care about